cache/class-vary-cache.php

```diff
@@ -426,11 +429,18 @@ private static function parse_group_cookie() {
 	 * @return string A string representation of the groups
 	 */
 	private static function stringify_groups() {
+		if ( empty( self::$groups ) ) {
+			return;
+		}
+
 		ksort( self::$groups ); // make sure the string order is the same every time.
-		$flatten = function ( $key, $value ) {
-			return $key . self::VALUE_SEPARATOR . $value;
-		};
-		$flattened = array_map( $flatten, array_keys( self::$groups ), self::$groups );
+		$flattened = [];
+		foreach ( self::$groups as $key => $value ) {
+			if ( '' === trim( $value ) ) {
+				continue;
+			}
+			$flattened[] = $key . self::VALUE_SEPARATOR . $value;
+		}
```

Comment:
To handle the case where I have groups registered but not actually set for a user (or set to an empty value).

```suggestion
		}

		if ( empty( $flattened ) ) {
			return;
		}
```